### ISSUE TITLE
Instability in GetCurvePointToWorldTransformAtPointIndex when curve points are too close to each other

### ISSUE THREAD

**Issue by cpinter** [Slicer/Slicer#5559]
When the control points of a curve are too close to each other, then the X/Y directions of the transform given by GetCurvePointToWorldTransformAtPointIndex are unstable regarding small changes in the curve.

A workaround is to reduce the number of curve points (= increase the distance between the curve points) by setting a lower number of points per segment (SetNumberOfPointsPerInterpolatingSegment).

A better fix would be to improve vtkFrenetSerretFrame so that it skips points until reaching a minimum distance between one point and the next considered point.

## Steps to reproduce

https://user-images.githubusercontent.com/1325980/113170498-857b2580-923e-11eb-9a1c-9955b86f0d1d.mp4

## Expected behavior

The orientation of the clip at the end of the device does not flicker.
In other words, the X/Y axes of the transform gven by GetCurvePointToWorldTransformAtPointIndex are more stable.

## Environment
- Slicer version: Slicer-4.13.0-2021-03-21
- Operating system: Windows 10

**Comment by cpinter** [comment]
@lassoan Please review and modify if necessary, thanks!

**Comment by lassoan** [comment]
I've implemented a more robust coordinate system computation (similarly to VMTK) - https://github.com/Slicer/vtkAddon/pull/19. Once it gets integrated, I'll update Slicer to use this class instead of Frenet-Serret frame.
